### PR TITLE
Flatten the dependency graph of Microsoft.AspNetCore.App

### LINKED ISSUE #3607: Produce platform-specific packages for Microsoft.AspNetCore.App
### Problem

Currently, framework-dependent deployments of Microsoft.AspNetCore.App use ready-to-run optimized images from the shared framework installation. Self-contained applications, however, do not get the ready-to-run optimizations. This happens because `dotnet publish` collects implementations binaries from NuGet packages and puts them in the publish directory, and the versions of ASP.NET Core assemblies in NuGet are not crossgen optimized.

### Proposed solution

Produce platform-specific packages for Microsoft.AspNetCore.App, and use this to provide ready-to-run optimized versions of .dlls for self-contained apps.

### Requirements

Follow the convention set by runtime.$(rid).Microsoft.NETCore.App packages. This means introducing:

* runtime.win-x64.Microsoft.AspNetCore.App
* runtime.win-x86.Microsoft.AspNetCore.App
* runtime.osx-x64.Microsoft.AspNetCore.App
* runtime.linux-x64.Microsoft.AspNetCore.App
* runtime.linux-arm.Microsoft.AspNetCore.App
* runtime.linux-musl-x64.Microsoft.AspNetCore.App

Package layout:
```
runtimes/
  $(rid)/
    lib/
       *.dll (ready-to-run optimized binaries)
    native/
       *.{so,dylib,dll} (native binaries, if any)
ref/
  netstandard/
    _._   (placeholder required to make NuGet happy and to ensure this package doesn't introduce compilation refs)
```

### PR DESCRIPTION
**Pull request by natemcmaster**
This changes the way Microsoft.AspNetCore.App works to follow patterns set by Microsoft.NETCore.App. Instead of being a metapackage with dozens of dependencies, this package has no dependencies. It uses RID-splitting to deliver standalone assets for self-contained deployments.

Changes:
* Implements RID-split packages for Microsoft.AspNetCore.App. Resolves #3607 
* Update shared fx deps.json generation to only include entries for the RID-specific metapackages
* Include platform-specific packages in publish output
* Remove all nuspec dependencies of Microsoft.AspNetCore.App and collect all references into the package. Resolves https://github.com/aspnet/AspNetCore/issues/3608.
  *Note*: the "reference assemblies" are actually the implementation assemblies. We'll replace these with true ref assemblies in a separate PR.

Notably, these related issues will be addressed in a separately https://github.com/aspnet/AspNetCore/issues/3609, https://github.com/aspnet/AspNetCore/issues/3610

FYI @dsplaisted @DustinCampbell @DamianEdwards

### CODE REVIEW

**Comment by natemcmaster** [comment]
Build passes in the PR. Tests fail due to disk-space limitations in Azure Pipelines. I've built and tested locally and believe this is ready to merge. I'd like to get this in soon so we can get this into the SDK for further testing of FrameworkReference.

File: build/artifacts.props [hunk 1]
```diff
@@ -23,7 +23,7 @@
     <PackageArtifact Include="Microsoft.AspNet.Identity.AspNetCoreCompat" Category="noship" />
     <PackageArtifact Include="Microsoft.AspNetCore.ApiAuthorization.IdentityServer" Category="noship" />
     <PackageArtifact Include="Microsoft.AspNetCore.App" Category="ship" />
-    <PackageArtifact Include="runtime.$(SharedFxRid).Microsoft.AspNetCore.App" Category="noship" />
+    <PackageArtifact Include="runtime.$(SharedFxRid).Microsoft.AspNetCore.App" Category="ship" />
```
Comment (muratg) [inline build/artifacts.props]: By design? Are we going to be shipping this as a package now?

**Review by muratg — APPROVED** [review]
I don't know the design that well, but the code looks good to me. Approving to unblock preview1 check-in.

File: build/artifacts.props (hunk 1 above)
Comment (natemcmaster) [reply build/artifacts.props]: Yup, this is intentional. This is how standalone deployment works, and what the "runtime.json" file is for.

**Pull request merged**